src/plugins/visualizations/public/plugin.ts

```diff
@@ -400,8 +402,31 @@ export class VisualizationsPlugin
     uiActions.registerTrigger(dashboardVisualizationPanelTrigger);
     const editInLensAction = new EditInLensAction(data.query.timefilter.timefilter);
     uiActions.addTriggerAction(CONTEXT_MENU_TRIGGER, editInLensAction);
-    const embeddableFactory = new VisualizeEmbeddableFactory({ start });
-    embeddable.registerEmbeddableFactory(VISUALIZE_EMBEDDABLE_TYPE, embeddableFactory);
+    embeddable.registerReactEmbeddableFactory(VISUALIZE_EMBEDDABLE_TYPE, async () => {
+      const {
+        plugins: { embeddable: embeddableStart, embeddableEnhanced: embeddableEnhancedStart },
+      } = start();
+
+      const { getVisualizeEmbeddableFactory } = await import('./react_embeddable');
+      return getVisualizeEmbeddableFactory({ embeddableStart, embeddableEnhancedStart });
+    });
+    embeddable.registerReactEmbeddableSavedObject<VisualizationSavedObjectAttributes>({
+      onAdd: (container, savedObject) => {
+        container.addNewPanel<VisualizeRuntimeState>({
+          panelType: VISUALIZE_EMBEDDABLE_TYPE,
+          initialState: savedObjectToRuntimeState(savedObject),
```

Comment:
Shouldn't `container.addNewPanel` get passed serializedState? Why are you converting the state to runtime state? What happens if you remove `savedObjectToRuntimeState`?

The reason I ask is that this PR adds 10KB to the page load size and am trying to track down all possible increases. It would be great if we could avoid importing `savedObjectToRuntimeState` into the page load bundle size. I don't think this is needed.

Reply:
Just tested it and the following works.

```
container.addNewPanel<VisualizeSerializedState>({
          panelType: VISUALIZE_EMBEDDABLE_TYPE,
          initialState: { savedObjectId: savedObject.id },
        });
```

Reply:
At one point this was necessary after a merge with `main` broke the embeddable, but I guess something with the panel handling changed and now it's no longer necessary? Still works without passing this in. I'll remove.